src/BloomExe/Program.cs

```diff
@@ -172,11 +171,20 @@ static void Main(string[] args1)
 							InstallerSupport.MakeBloomRegistryEntries();
 							Browser.SetUpXulRunner();
 							Browser.XulRunnerShutdown += OnXulRunnerShutdown;
-							L10NSharp.LocalizationManager.SetUILanguage(Settings.Default.UserInterfaceLanguage, false);
+							LocalizationManager.SetUILanguage(Settings.Default.UserInterfaceLanguage, false);
 							var transfer = new BookTransfer(new BloomParseClient(), ProjectContext.CreateBloomS3Client(),
 								_applicationContainer.HtmlThumbnailer, new BookDownloadStartingEvent())/*not hooked to anything*/;
 							transfer.HandleBloomBookOrder(args[0]);
 							PathToBookDownloadedAtStartup = transfer.LastBookDownloadedPath;
+
+							// BL-2143: Show failure message if download was not successful
+							if (string.IsNullOrEmpty(PathToBookDownloadedAtStartup))
+							{
+								var errCaption = LocalizationManager.GetString("Download.FailedCaption", "Download not completed");
+								var errMessage = LocalizationManager.GetString("Download.Failed", "Your download from BloomLibrary.org did not complete.");
+								MessageBox.Show(errMessage, errCaption, MessageBoxButtons.OK, MessageBoxIcon.Exclamation);
+							}
```

Comment:
In Len's case, there was already a message indicating (more about) what went wrong...I don't think we would need this as well. Is there a failure path where download fails and this is the only indication? If not I would be inclined not to bother them with another message. But it might be worth checking with @hatton.